src/main/java/com/daitangroup/vehiclesTypes/Car.java

```diff
@@ -0,0 +1,232 @@
+package com.daitangroup.vehiclesTypes;
+
+import com.daitangroup.exceptions.EngineException;
+import com.daitangroup.interfaces.Vehicle;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class Car implements Vehicle {
+
+    /**
+     * slf4j var for logging
+     */
+    final static Logger LOGGER = LoggerFactory.getLogger(Car.class);
+
+
+    /**
+     * Attributes for Car
+     */
+    private float fuel;
+    private int numberOfGears;
+    private int actualGear;
+    private float loadCapacity;
+    private int numberOfDoors;
+    private int maxNumberOfOccupants;
+    private Set<String> occupantsNames = Collections.emptySet();
+    private boolean engineIsOn;
+
+
+    //Implementation for vehicle interface
+    public void accelerateVehicle() {
+        System.out.println("Vrum");
+    }
+
+    public void breakVehicle() {
+        System.out.println("Break");
+    }
+
+    public void turnLeft() {
+        System.out.println("Turning Left");
+    }
+
+    public void turnRight() {
+        System.out.println("Turning Right");
+    }
+
+    public float checkFuel() {
+
+        return fuel;
+    }
+
+    public void turnVehicleOn() throws EngineException {
+        System.out.println("Vehicle Turned On");
+        if(engineIsOn) {
+            throw new EngineException("Engine is already on.");
+        } else {
+            engineIsOn = true;
+            System.out.println("Vehicle Turned On");
+        }
+    }
+
+    public void turnVehicleOff() throws EngineException {
+        if(!engineIsOn) {
+            throw new EngineException("Engine is already Off.");
+        } else {
+            engineIsOn = false;
+            System.out.println("Vehicle Turned Off");
+        }
+    }
+
+    public int changeToNextGear() {
+        if(actualGear < numberOfGears) {
+            actualGear++;
+            LOGGER.warn("Changing gear.");
+        } else {
+            LOGGER.warn("No more gears left to change");
+        }
+        return actualGear;
+    }
+
+    public int changeToPreviousGear() {
+        if(actualGear > 1) {
+            actualGear--;
+            LOGGER.warn("Changing gear.");
+        } else {
+            LOGGER.warn("No more gears left to change");
+        }
+        return actualGear;
+    }
+
+    public void putOnNeutral() {
+        actualGear = 0;
+        System.out.println("Putting on neutral");
+    }
+
+
+    /**
+     * Class constructor
+     * @param fuel Fuel of car in liters
+     * @param numberOfGears Number of shift gears
+     * @param loadCapacity Trunk load capacity
+     * @param numberOfDoors Numeber of doors in the car
+     * @param maxNumberOfOccupants Maximum number of occupants
+     */
+    public Car(float fuel, int numberOfGears, float loadCapacity, int numberOfDoors, int maxNumberOfOccupants) {
```

Comment:
Just to better readability and convention, the constructor should the first method.